### PR TITLE
Add url parameter to configure the queueUrl of an SqsEndpoint

### PR DESCRIPTION
**Pull request by DaniRey**
I think it would be preferable to configure the queueUrl of an SqsEndpoint, to allow for integration testing against a mock implementation of AWS SQS, such as ElasticMQ.

To do so the most important things to be able to change are the protocol (http/https) and the host (localhost/sqs.{region}.amazonaws.com). In my opinion being able to control the complete queueUrl via urlParam is preferable over just controlling protocol and host, because it is easier to understand and it provides additional flexibility.

### CODE REVIEW

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsEndpoint.java [hunk 1]
```diff
@@ -121,28 +121,30 @@ protected void doStart() throws Exception {
...
-            for (String url : listQueuesResult.getQueueUrls()) {
-                if (url.endsWith("/" + configuration.getQueueName())) {
-                    queueUrl = url;
-                    LOG.trace("Queue available at '{}'.", queueUrl);
-                    break;
-                }
+            // If both region and Account ID is provided the queue URL can be built manually.
+            // This allows accessing queues where you don't have permission to list queues or query queues
+            if (configuration.getRegion() != null && configuration.getQueueOwnerAWSAccountId() != null) {
+                queueUrl = "https://sqs." + configuration.getRegion() + ".amazonaws.com/"
```
Comment (oscerd) [inline components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsEndpoint.java]: Why did you remove the host specification? Can you add it again?

Comment (DaniRey) [reply components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsEndpoint.java]: Sorry, I was not careful enough. We're still on 2.18.4, so my changes stem from this version and I missed that improvement. I will put the host back in.

**Comment by DaniRey** [comment]
`mvn clean install -Psourcecheck` didn't throw any errors, but I realized that I commited Tabs instead of Spaces. I fixed this with my last commit. I hope this where the codestyle issues, which concerned you.

**Comment by oscerd** [comment]
Looks good, I'll merge asap :-) Thanks!

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java [hunk 2]
```diff
@@ -63,6 +63,8 @@
     private boolean extendMessageVisibility;
     @UriParam(label = "consumer", defaultValue = "1")
     private int concurrentConsumers = 1;
+    @UriParam
```
Comment (davsclaus) [inline components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java]: This should be labelled as advanced. As its not a common thing to use.

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java [hunk 3]
```diff
@@ -345,6 +347,17 @@ public int getConcurrentConsumers() {
     public void setConcurrentConsumers(int concurrentConsumers) {
         this.concurrentConsumers = concurrentConsumers;
     }
+	
+    /**
+     *  To define the queueUrl explicitly. All other parameters, which would influence the queueUrl, are ignored.
+     */
```
Comment (davsclaus) [inline components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java]: Add better javadoc what the purpose of this is, eg for testing/mocking etc. That it should not be used normally.

**Comment by oscerd** [comment]
I'll wait to merge then

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java (hunk 2 above)
Comment (DaniRey) [reply components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java]: I was not aware of this possibility, that's a lot better.

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java (hunk 3 above)
Comment (DaniRey) [reply components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java]: I tried to improve this. I hope the new text covers the purpose well enough.

**Comment by DaniRey** [comment]
Thanks for those speedy replies from the two of you!

I totally see your point @davsclaus. I initially tried to do it without any code changes.

Without being able to switch SSL off, the only solution I could come up with is to configure a proxy for SQS and register self-signed certificates for "amazonaws.com". Also the testing infrastructure would need at least three systems: a proxy, SSL termination, the mock itself. In my opinion this is to complex to be feasible.

Therefore I decided that changing camel-aws is the better option. I initially started with making it possible to control whether to use SSL or not. After working with this, I decided that providing full control over the queueUrl is prefered for the following reasons: 
 - when working with AWS SQS people should NEVER switch to http
 - when working with a mock having a single configuration to change the complete URL is easier to understand
 - it allows users to cover unforseen changes in AWS URLs, such as not having the "sqs." prefix anymore

**Comment by davsclaus** [comment]
@DaniRey thanks for the explanation, it makes sense now. @oscerd I think its good to merge.

**Comment by oscerd** [comment]
yeah, I'll merge this afternoon. Thanks @DaniRey and @davsclaus

**Comment by oscerd** [comment]
Thanks, the PR has been merged on master branch.
